src/assets/highlight-links.js

```diff
@@ -0,0 +1,54 @@
+// eslint-disable-next-line spaced-comment
+/// <reference types="chrome"/>
+
+// Listens for a request from LinkService and respondes with all of the links
+// from the contents of the active page.
+chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
+  try {
+    if (request.message === 'highlight link') {
+      sendResponse({
+        success: true,
+        highlightOn: highlightLink(request.linkData)
+      });
+    }
+  } catch (e) {
+    console.error(e);
+    sendResponse({
+      success: false,
+      errorMessage: 'Error highlighting links: ' + e.message
+    });
+  }
+
+});
+
+function highlightLink(link) {
+  const matches = document.getElementsByClassName(link.highlightId);
+
+  if (matches.length === 0) {
+    alert('Could not find the selected link on the original page. ' +
+        'Link Diver\'s data could be out of date, ' +
+        'please try reloading Link Diver to get the most up to date links.');
+    throw new Error('selected link could not be found on the page');
+  }
+
+  const element = matches[0];
+  const highlightClass = 'link-diver-highlight';
+
+  const isHighlighted = element.classList.contains(highlightClass);
+  if (isHighlighted) {
+    element.classList.remove(highlightClass);
+  } else {
+    element.classList.add(highlightClass);
+    element.scrollIntoView({
```

Comment:
Between scrollIntoView() and focus(), I'm not really sure which one is better. I think focus might be slightly more reliable, but the smooth scrolling is really nice.

Reply:
After some more investigating I've found situations one works and the other doesn't so maybe we should include them both?

Reply:
Do you know why it happened in those cases?

Reply:
I think focus() doesn't always work because not all of the links we are extracting are focusable. ScrollIntoView() seems to be less clear, but the cases I found seemed to be due to the page not wanting to scroll through the footer